kaspad/src/daemon.rs

```diff
@@ -357,7 +357,7 @@ do you confirm? (answer y/n or pass --yes to the Kaspad command line to confirm
 
     let tick_service = Arc::new(TickService::new());
     let (notification_send, notification_recv) = unbounded();
-    let max_tracked_addresses = if args.utxoindex { args.max_tracked_addresses } else { None };
+    let max_tracked_addresses = if args.utxoindex && args.max_tracked_addresses > 0 { Some(args.max_tracked_addresses) } else { None };
```

Comment:
Make it Option<NonZeroUsize>. It has the same layout as usize, and better reflects purpose of its use

Reply:
It's a cool structure to know about, but I'm not sure it's needed here (should definitely be used when a vector of such values is created to reduce mem, but here it's over-fancy imho).